vernac/metasyntax.ml

```diff
@@ -1002,8 +1002,13 @@ let find_precedence lev etyps symbols =
   | None -> [],0
   | Some (NonTerminal x) ->
       (try match List.assoc x etyps with
-	| ETConstr _ ->
-	    user_err Pp.(str "The level of the leftmost non-terminal cannot be changed.")
+	| ETConstr (clev,_) ->
+	   if onlyprint then
+	     match clev with
+	     | Extend.NextLevel -> user_err Pp.(str "The level of the leftmost non-terminal must be a specific number, not the next level.")
+	     | Extend.NumLevel n -> [],n
```

Comment:
This is not right. What is returned on the right is the level of the notation, not the level of its first subentry. The code should be something like:
```coq
| ETConstr _ ->
    if onlyprint then
      if Option.is_empty lev then
        user_err Pp.(str "Explicit level needed in only-printing mode when the level of the leftmost non-terminal is given.")
      else [],Option.get lev
    else
      user_err Pp.(str "The level of the leftmost non-terminal cannot be changed.")

```

Reply:
In fact, that's what I had tried at first, because that was my (somewhat shaky) understanding of the code. But it produces different results. For the two output tests, the test-suite log shows:

```
-(*1).2.3
+((*1).2).3
      : nat
-*1.2
+*(1.2)
      : nat
```
If the alternative approach mentioned in the bug report is satisfactory, maybe we should close this PR.

Reply:
This is expected: in the first example, you see a `plus` so, the `.` rule applies and expects its right argument at level 14. Then, another `plus` is found. Its level is 20 which is higher than the 14 which is accepted, so a parenthesis is added. Idem when seeing `id`. The rule for `*` is at level 15 >= 14, so a parenthesis is added so as to reinitialize the current level at 200 and to be able to write `*`.

In the second example, there is no level given to the argument of the rule for `id`. By default, it is right associative, so it will be at level 15. But this argument is a `plus` which has a notation at level 20. To be able to use this notation, a parenthesis is inserted to reinitialize the current level to its higher level 200.